### ISSUE TITLE
Rework Issue creation templates

### ISSUE THREAD

**Issue by objecttothis** [opensourcepos/opensourcepos#3895]
Currently we use a text-based template that we ask people to fill out when creating an issue. We need to instead create a github issue template for each type of issue (https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository)

Doing this will reduce how often we have to ask people to provide more details.  Some of the fields need to be required so that people can not submit bug reports without important information.

The types that I can think of are:
Bug Report
Feature Request
Question

@odiea do you feel like taking this one on?

**Comment by odiea** [comment]
I will look into this after the first of the new year.

**Comment by objecttothis** [comment]
@odiea do you still plan on looking at this? No worries if not. I'm just curious on the status.

**Comment by odiea** [comment]
If there is any coding involved I won't be able to help out. Sorry

**Comment by objecttothis** [comment]
I wouldn't call it coding per se.  It involves writing a YAML markdown "template" (https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository). I'd be willing to bet that with a quick google search you'd probably even be able to find pre-made templates that people have already done and just tweak them to meet our needs.  Take a look and then you can decide.  No worries if it feels too much.

**Comment by objecttothis** [comment]
This guy has a repo of them, so maybe you could just take a ready-made one and not really need to do much of anything?

https://github.com/stevemao/github-issue-templates

**Comment by odiea** [comment]
Here is what I have learned so far. The feature request is an option under Issues. See Image. The first image is clicking on issues. The second is selecting the feature request. here is my repository if you would like to try it out.
![2024-02-19 13_17_47-New Issue · odiea_opensourcepos](https://github.com/opensourcepos/opensourcepos/assets/10962177/0127a453-186d-423a-801a-ee3c09c2b865)

![2024-02-19 13_16_55-New Issue · odiea_opensourcepos](https://github.com/opensourcepos/opensourcepos/assets/10962177/2d7bac4c-211e-419f-8bff-885e96172511)

**Comment by jekkos** [comment]
Looks very clean and structured! Maybe one last item for bug reports. 
Also I wonder if we can label the issues automatically depending on the submission type.

**Comment by odiea** [comment]
Ok. I just added the bug report template. Go to issues click issues again then the bug report template should show. The also I don't know or did not find any info on.

**Comment by odiea** [comment]
I think this is what you are looking for. Choose bug issue.

**Comment by objecttothis** [comment]
> I think this is what you are looking for. Choose bug issue.

Yep. Perhaps the code of conduct checkbox could be a confirmation they are submitting a bug found in an unmodified release of OpensourcePOS. We might need an "other" option for the browser. Also, IMO we need dropdowns for their PHP version, Server OS/version textbox, DBMS and http server flavor drop-down.

**Comment by odiea** [comment]
You can see by the number of commits how much fun I had changing this template.

**Comment by objecttothis** [comment]
> You can see by the number of commits how much fun I had changing this template.

Yes you did! :) Good work.

Here are the further changes I would make to "Bug Issue"
- Rename from "Issue: Bug Issue" to "Bug Report"
- Change the introduction to something like "Bug reports indicate that something is not working as intended. Please include as much detail as possible and submit a separate bug report for each problem. Do not include personal identifying information such as email addresses or encryption keys."
- Break out the "What Happened?" section into the following sections
- ----"Bug Description" with a subtitle of "Describe the problem that you are seeing."
- ---- "Expected Behavior" with the subtitle of "Describe what behavior you expect."
- ---- "Steps to Reproduce" with the subtitle of "List the detailed steps taken to reproduce this problem. Include screenshots where possible."
- The "Version" section should be the release of OSPOS rather than the CodeIgniter framework version.  If possible to put the dropdown to the right of the word "Version" that would save space. Also, I would change it from "Version" to "OpensourcePOS Version". I wonder if there is a way to get the template to automatically pull from the list of releases instead of having to adjust the template each time there is a release.
- The PHP version should maybe have the subtitle "If "other" please list in bug description, but the dropdown should include PHP 7.4, 8.0, 8.1, 8.2, 8.3, and other.
- I would make a separate textbox for "Web Server and Version" but just make it one line and maybe 150px wide?
- The "Required Info?" section could maybe be renamed to "System Information Report (optional)" with a subtitle "Copy and paste from OSPOS > Configuration > Setup & Conf > Setup & Conf"
- I would get rid of the heading Code of Conduct and it's subtitle then just leave the checkbox and reword it to "This bug occurs on an unmodified version of OpensourcePOS"
- I would move all the checkboxes, dropdowns and small questions to the beginning of the bug report.

**Comment by odiea** [comment]
I am finding that I do not have the patience for this. I will delete it after your next commit. Sorry

**Comment by objecttothis** [comment]
OK, I created a branch and a draft pull request.  I'll take a look when I get a chance.  Or someone else can take it on.

**Comment by odiea** [comment]
I had more patience this morning so it is working better. Check it out so that I can add more info where needed

**Comment by objecttothis** [comment]
Your patience paid off. It looks fantastic!  Is it possible to add three more text boxes only small (not like the text areas that we currently have).
- Server Operating System and version
- Database Management System and version
- Web Server and version

The reason for breaking it out is that the Configuration report is optional but those three pieces of information is sometimes super helpful so we need to collect those even if the submitter doesn't copy/paste the report... sometimes they can't get to the point where the report is displaying.

**Comment by odiea** [comment]
Once I get the damn thing working again I will let you know

**Comment by objecttothis** [comment]
For feature request you might be able to copy something like this and just tweak it https://github.com/cheehwatang/.github/blob/master/.github/ISSUE_TEMPLATE/feature_request.yml

**Comment by odiea** [comment]
I will look into the template. I think that I have done enough damage on the bug template for now. LOL

**Comment by odiea** [comment]
I added this template and just changed the assignee for now. Give recommendations as needed.

**Comment by odiea** [comment]
Will we need a disclaimer on which versions of OSPOS would receive Feature Requests?

**Comment by objecttothis** [comment]
> Will we need a disclaimer on which versions of OSPOS would receive Feature Requests?

That's a good idea. Perhaps @jekkos has thoughts, but other than seeing if we can easily make 3.3.9 PHP 8.0 compatible, we should feature-lock it at 3.3.10, then only future releases receive feature changes.

**Comment by odiea** [comment]
I added the ospos version dropdown to the version request

**Comment by jekkos** [comment]
Very nice, this is very helpful. As a last thing I would add a checkbox at the very end of the feature request form to verify that the user did search open requests in the tracker. The text is already there but I'd add a checkbox to it.
